phpBB/phpbb/avatar/driver/upload.php

```diff
@@ -151,13 +167,35 @@ public function process_form($request, $template, $user, $row, &$error)
 			$destination = '';
 		}
 
-		// Move file and overwrite any existing image
-		$file->move_file($destination, true);
+		/**
+		* Before overwriting an existing avatar with a newly uploaded avatar
+		*
+		* @event core.avatar_driver_upload_overwrite_before
+		* @var	string	destination			Destination directory where the file is going to be moved
+		* @var	string	prefix				Prefix for the avatar filename
+		* @var	array	row					Array with avatar row data
+		* @var	array	error				Array of errors, if filled in by this event file will not be moved
+		* @since 3.1.6-RC1
+		*/
+		$vars = array(
+			'destination',
+			'prefix',
+			'row',
+			'error',
+		);
+		extract($this->dispatcher->trigger_event('core.avatar_driver_upload_overwrite_before', compact($vars)));
```

Comment:
I'd like it more if you could change the name to core.avatar_driver_upload_move_file_before. It's not always overwriting a file so calling it that is somewhat not correct.